test/test_content_store.py

```diff
@@ -60,6 +60,13 @@ def test_scalar(self, device):
         # Should not raise an error
         hash_storage(torch.tensor(2, device=device).untyped_storage())
 
+    @torch._dynamo.config.patch(cache_size_limit=1)
+    def test_repeated_hash(self, device):
+        # Test that repeated hashing doesn't trigger a recompile in dynamo
+        # If it does, we will execute prims.xor_sum in eager which fails
+        for _ in range(4):
+            hash_storage(torch.tensor(2, device=device).untyped_storage())
```

Comment:
FWIW, recreating the same closure in a loop shouldn't actually cause recompile if the code object stays the same and there are no free vars. cc @Fidget-Spinner who had a patch to fix this but I can't remember if it actually landed

Reply:
This test was failing for me on a fairly recent main branch commit.

Reply:
I just re-landed that patch in main two days ago https://github.com/pytorch/pytorch/pull/109427

Reply:
I just checked and cherry-picking that commit does make the test pass, oh well.

Reply:
It's fine, lazy compile at top level is a useful thing to have anyway